usecases/backup/coordinator.go

```diff
@@ -487,12 +522,31 @@ func (c *coordinator) abortAll(ctx context.Context, req *AbortRequest, nodes map
 }
 
 // groupByShard returns classes group by nodes
-func (c *coordinator) groupByShard(ctx context.Context, classes []string) (nodeMap, error) {
+func (c *coordinator) groupByShard(ctx context.Context, classes []string, leader string) (nodeMap, error) {
 	nodes := c.nodeResolver.AllNames()
 	m := make(nodeMap, len(nodes))
-	for _, node := range nodes {
-		m[node] = &backup.NodeDescriptor{Classes: slices.Clone(classes)}
+
+	// start by collecting nodes which have content
+	for _, cls := range classes {
+		nodes, err := c.selector.Shards(ctx, cls)
+		if err != nil {
+			continue
+		}
+		for _, node := range nodes {
+			if node == leader {
+				continue
+			}
+			nd, ok := m[node]
+			if !ok {
+				nd = &backup.NodeDescriptor{Classes: make([]string, 0, 5)}
```

Comment:
question(non-blocking): What is the reason we initialize this to 5 capacity ? Looking at the code we just append one element, so I think initializing to 1 would save memory.

Reply:
The `NodeDescriptor` is added once, while multiple classes might be added. A node might store more than once class. Using 5 as initial estimate helps prevents reallocation if the number of classes exceeds the threshold